src/harm.cpp

```diff
@@ -209,4 +210,47 @@ int Harm::AdjustHarmGrpsSpacing(FunctorParams *functorParams)
     return FUNCTOR_SIBLINGS;
 }
 
+int Harm::Transpose(FunctorParams *functorParams)
+{
+    TransposeParams *params = dynamic_cast<TransposeParams *>(functorParams);
+    assert(params);
+
+    LogDebug("Transposing harm");
+
+    Text *textObject = dynamic_cast<Text *>(this->GetChild(0, TEXT));
+    std::wstring text = textObject->GetText();
+
+    // Transpose chord names: "C".
+    if (text[0] >= 'A' && text[0] <= 'G') {
+        int accid = 0;
+        int i;
+        for (i = 1;; i++) {
+            switch (text[i]) {
+                case L'𝄫': accid -= 2; break;
+                case 'b':
+                case L'♭': accid--; break;
+                case '#':
+                case L'♯': accid++; break;
+                case L'𝄪': accid += 2; break;
+                default: goto break2;
```

Comment:
Please make sure there is no use of `goto`. Also, it is not a good idea to loop over the characters of the `text` variable and to change it within the loop. This needs to be refactored.

Reply:
I wanted to use `break 2;`, but C++ doesn't support that. https://stackoverflow.com/questions/1257744/can-i-use-break-to-exit-multiple-nested-for-loops recommends goto. (Also the text variable is not being changed within the loop).

Since this was confusing, I agree that it should be refactored.